platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -63,18 +76,287 @@
   private final int maximumShardsPerNode =
       NumberUtils.toInt(System.getProperty("solr.cloud.maxShardPerNode"), 2);
 
-  @Override
-  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
-    String zookeeperHosts = System.getProperty("solr.cloud.zookeeper");
+  private final Map<String, Integer> shardCountMap = new HashMap<>();
 
-    if (StringUtils.isBlank(zookeeperHosts)) {
+  private final Map<String, Integer> replicationFactorMap = new HashMap<>();
+
+  private final Map<String, Integer> maximumShardsPerNodeMap = new HashMap<>();
+
+  private String zookeeperHosts;
+
+  public SolrCloudClientFactory() {
+    String collectionList = System.getProperty("solr.cloud.collections");
+    if (StringUtils.isBlank(collectionList)) {
       LOGGER.warn(
-          "Cannot create Solr Cloud client without Zookeeper host list system property [solr.cloud.zookeeper] being set.");
-      throw new IllegalStateException("system property 'solr.cloud.zookeeper' is not configured");
```

Comment:
why do we no longer throw here?

Reply:
it is now throwing here https://github.com/codice/ddf/pull/5813/files/70f163717afc9e63491dc377c36ba18cd6318e3f#diff-d4aceb9ea86aa85e45c86639b76659e5R600